lib/rules-generator.js

```diff
@@ -405,24 +472,37 @@ util.methods(Generator, {
         if (replaceExp()) {
           exp.ref = ref;
           exp.args = args;
+          if (exp.ref.valueType == 'Snapshot') {
+            exp.valueType = 'Snapshot';
+          }
         }
       }
-      break;
-    }
 
-    return exp;
+      return exp;
+
+    default:
+      return exp;
```

Comment:
can we have an explanation of what the default case does? If its never supposed to run can we have a throw?